Sources/WalletConnectSign/Engine/Common/ApproveEngine.swift

```diff
@@ -277,7 +282,21 @@ private extension ApproveEngine {
             return respondError(payload: payload, reason: .invalidUpdateRequest, protocolMethod: SessionProposeProtocolMethod())
         }
         proposalPayloadsStore.set(payload, forKey: proposal.proposer.publicKey)
-        onSessionProposal?(proposal.publicRepresentation(pairingTopic: payload.topic))
+        
+        if let rawRequest = payload.rawRequest, let verifyClient {
+            Task(priority: .high) {
+            let attestationId = rawRequest.rawRepresentation.sha256().toHexString()
+                let origin = try? await verifyClient.registerAssertion(attestationId: attestationId)
```

Comment:
same here, shouldn't you call verifyOrigin() method on verify client?
I can see it's not called anywhere 🤔